File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
def solution(dartResult):
    score = ""
    answer = []
    SDTObj = {"S": 1,"D": 2,"T": 3}
    
    for v in dartResult:
        if v.isdigit(): score += v
        elif v.isalpha(): 
            answer.append(int(score) ** SDTObj[v])
            score = ""
        elif v == "*":
            if len(answer) > 1: answer[-2] *= 2
            answer[-1] *= 2
        elif v == "#": answer[-1] *= -1
    
    return sum(answer)

File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.js
function solution(dartResult) {
    let answer = [];
    let score = "";
    const SDTObj = {
        "S": 1,
        "D": 2,
        "T": 3
    }
    
    for (let v of dartResult){
        if (!isNaN(v)) score += v
        else if (/S|D|T/.test(v)) {
            answer.push(parseInt(score) ** SDTObj[v])
            score = ""
        } else if (v === "*") {
            if (answer.length > 1) answer[answer.length-2] *= 2
            answer[answer.length-1] *= 2
        } else if (v === "#") answer[answer.length-1] *= -1
    }
    return answer.reduce((a,c) => a + c, 0);
}